### PR TITLE
delete vae and text encoders after use in SDXL training script

### PR DESCRIPTION
**Pull request by noskill**
# What does this PR do?

Vae and encoders are kept in memory by partial function, which prevents SDXL training with 24gb of gpu memory.
creating and moving vae and encoders to gpu allocates approximately 3.25 GB. With this change 2 gb are freed.

### CODE REVIEW

File: examples/text_to_image/train_text_to_image_sdxl.py [hunk 1]
```diff
@@ -572,6 +572,15 @@ def generate_timestep_weights(args, num_timesteps):
     return weights
 
 
+def print_gpu_utilization(idx=0):
```
Comment (yiyixuxu) [inline examples/text_to_image/train_text_to_image_sdxl.py:575]: we probably don't need to add this function, no?

File: examples/text_to_image/train_text_to_image_sdxl.py [hunk 2]
```diff
@@ -699,7 +709,7 @@ def main(args):
 
             xformers_version = version.parse(xformers.__version__)
             if xformers_version == version.parse("0.0.16"):
-                logger.warn(
+                logger.warning(
```
Comment (sayakpaul) [inline examples/text_to_image/train_text_to_image_sdxl.py:712]: Why this change? We use `warn` throughout our training scripts.

File: examples/text_to_image/train_text_to_image_sdxl.py [hunk 3]
```diff
@@ -892,6 +902,7 @@ def preprocess_train(examples):
             new_fingerprint=new_fingerprint_for_vae,
         )
```
Comment (sayakpaul) [inline examples/text_to_image/train_text_to_image_sdxl.py:904]: Makes sense.

**Review by sayakpaul — COMMENTED** [review]
Hi,

Thanks for your contributions. Let’s resolve the comments. And we can then merge.

**Comment by sayakpaul** [comment]
Also apologies for the delay here.

File: examples/text_to_image/train_text_to_image_sdxl.py (hunk 1 above)
Comment (noskill) [reply examples/text_to_image/train_text_to_image_sdxl.py]: yes

**Comment by sayakpaul** [comment]
Ping me when the CI run is complete.

**Comment by noskill** [comment]
@sayakpaul ping
ci run is complete

**Comment by noskill** [comment]
@sayakpaul ping
